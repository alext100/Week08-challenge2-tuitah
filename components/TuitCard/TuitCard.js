import "./TuitCard.module.css";
import TimeAgo from "javascript-time-ago";
import es from "javascript-time-ago/locale/es.json";
import ReactTimeAgo from "react-time-ago";

TimeAgo.addLocale(es);

const TuitCard = ({ tuit }) => (
  <>
    <article className="tuit-card col-md-5 col-12 row justify-content-center">
      <div className="col-12 row">
        <p className="tuit-card__text col-12">{tuit.text}</p>
        <p className="tuit-card__date col-md-9 col-12">
          <ReactTimeAgo date={Date.parse(tuit.date)} locale="es" />
        </p>
        <p className="tuit-card__likes col-md-3 col-12">Likes: {tuit.likes}</p>
      </div>
      <div className="col-12 row">
        <button type="button" className="btn btn-dark col-md-6 col-12">
          Like
        </button>
        <button type="button" className="btn btn-danger col-md-6 col-12">
          Delete
        </button>
      </div>
    </article>
  </>
);

export default TuitCard;
