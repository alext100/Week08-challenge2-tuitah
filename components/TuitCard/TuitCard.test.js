import ReactTestRenderer from "react-test-renderer";
import TuitCard from "./TuitCard";

describe("Given a TuitCard component", () => {
  describe("When it's rendered", () => {
    test("Then it should show match the last snapshot", () => {
      const tuit = {
        id: "619688c43b799656d5806576",
        text: "ppppppppppppppppppp",
        likes: 0,
        date: "2021-11-18T17:09:24.499Z",
      };
      const tuitCard = ReactTestRenderer.create(
        <TuitCard key={tuit.id} tuit={tuit} />
      );

      expect(tuitCard.toJSON()).toMatchSnapshot();
    });
  });
});
