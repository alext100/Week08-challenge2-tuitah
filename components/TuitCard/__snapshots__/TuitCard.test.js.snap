// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a TuitCard component When it's rendered Then it should show match the last snapshot 1`] = `
<article
  className="tuit-card col-md-5 col-12 row justify-content-center"
>
  <div
    className="col-12 row"
  >
    <p
      className="tuit-card__text col-12"
    >
      ppppppppppppppppppp
    </p>
    <p
      className="tuit-card__date  col-md-9 col-12"
    >
      <time
        dateTime="2021-11-18T17:09:24.499Z"
      >
        hace 4 horas
      </time>
    </p>
    <p
      className="tuit-card__likes col-md-3 col-12"
    >
      Likes: 
      0
    </p>
  </div>
  <div
    className="col-12 row"
  >
    <button
      className="btn btn-dark col-md-6 col-12"
      onClick={[Function]}
      type="button"
    >
      Like
    </button>
    <button
      className="btn btn-danger col-md-6 col-12"
      onClick={[Function]}
      type="button"
    >
      Delete
    </button>
  </div>
</article>
`;
